test/integration/scheduler/util.go

```diff
@@ -548,3 +548,18 @@ func nextPodOrDie(t *testing.T, testCtx *testutils.TestContext) *framework.Queue
 	}
 	return podInfo
 }
+
+// nextPod returns the next Pod in the scheduler queue, with a 15 seconds timeout.
+func nextPod(t *testing.T, testCtx *testutils.TestContext) *framework.QueuedPodInfo {
+	t.Helper()
+
+	var podInfo *framework.QueuedPodInfo
+	// NextPod() is a blocking operation. Wrap it in timeout() to avoid relying on
+	// default go testing timeout (10m) to abort.
+	if err := timeout(testCtx.Ctx, time.Second*15, func() {
```

Comment:
I wonder whether this can be refactored as:
```
	timeout(testCtx.Ctx, time.Second*15, func() {
		podInfo = testCtx.Scheduler.NextPod()
	})
	return podInfo
```

since error msg will not be used.

Reply:
your code implementation is more clearly, forget it.